transport/grpc/outbound_test.go

```diff
@@ -200,3 +202,82 @@ func TestCallStreamWithInvalidPeer(t *testing.T) {
 		}.Error(),
 	)
 }
+
+func TestCallSvcMatch(t *testing.T) {
+	tests := []struct {
+		msg         string
+		serverPort  string
+		headerKey   string
+		headerValue string
+		wantErr     bool
+	}{
+		{
+			msg:         "test call service match success",
+			serverPort:  "15213",
+			headerKey:   ServiceHeader,
+			headerValue: "Service",
+		},
+		{
+			msg:         "test call service match failed",
+			serverPort:  "15619",
+			headerKey:   ServiceHeader,
+			headerValue: "ThisIsWrongSvcName",
+			wantErr:     true,
+		},
+		{
+			msg:        "test no service name response header",
+			serverPort: "15640",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.msg, func(t *testing.T) {
+			server := grpc.NewServer(
+				grpc.CustomCodec(customCodec{}),
+				grpc.UnknownServiceHandler(func(srv interface{}, stream grpc.ServerStream) error {
+					responseWriter := newResponseWriter()
+					defer responseWriter.Close()
+
+					responseWriter.AddSystemHeader(tt.headerKey, tt.headerValue)
```

Comment:
it may be worth it to check if `tt.headerKey` is empty we shouldn't write anything

Reply:
Fixed.